packages/domain/lib/use_case/get_movies_use_case.dart

```diff
@@ -1,22 +1,34 @@
 import 'package:domain/enum/movie_type.dart';
 import 'package:domain/model/data_model.dart';
+import 'package:domain/model/movie_db_model.dart';
 import 'package:domain/model/movie_response.dart';
+import 'package:domain/repository/movie_database_local_repository.dart';
+import 'package:domain/repository/preference_local_repository.dart';
 import 'package:domain/repository/trakt_repository.dart';
 import 'package:domain/use_case/use_case.dart';
 import 'package:domain/utils/const.dart';
 
 class GetMoviesUseCase
-    implements UseCaseParams<MovieType, Future<List<MovieResponse>>> {
-  const GetMoviesUseCase(this._repository);
+    implements UseCaseParams<MovieType, Future<List<MovieDBModel>>> {
+  const GetMoviesUseCase(
+    this._traktRepository,
+    this._localRepository,
+    this._preferences,
+  );
 
-  final TraktRepository _repository;
+  final TraktRepository _traktRepository;
+  final MovieDBLocalRepository _localRepository;
+  final PreferencesLocalRepository _preferences;
 
   @override
-  Future<List<MovieResponse>> call(MovieType type) async {
+  Future<List<MovieDBModel>> call(MovieType type) async {
     final List<MovieResponse> jsonMovies = [];
-    final response = type == MovieType.trending
-        ? await _repository.getDataTrending()
-        : await _repository.getDataAnticipated();
+    final List<MovieDBModel> cachedMovies =
+        await _localRepository.getMovieFromCache(type);
+
+    final GetDataResponse response = type == MovieType.trending
```

Comment:
before make call to api or fetching from cache u should to know last api call date, then compare with current date and decide to use cached movies or do api call.